server/game/cards/characters/01/oldforesthunter.js

```diff
@@ -4,30 +4,16 @@ class OldForestHunter extends DrawCard {
     setupCardAbilities(ability) {
         this.action({
             title: 'Discard a card to gain 1 gold',
-            method: 'discard',
-            limit: ability.limit.perPhase(1)
+            condition: () => !this.controller.cannotGainGold,
+            cost: ability.costs.discardFromHand(),
+            limit: ability.limit.perPhase(1),
+            handler: context => {
+                this.game.addGold(this.controller, 1);
```

Comment:
Re: conflicting messages, I've thought about converting some methods to promises or promise-like syntax so that in cases where an action doesn't succeed (card didn't get put into play as expected, gold wasn't gained in this case), the message doesn't go out. Something like:

```javascript
this.game.addGold(this.controller, 1).then(gold => {
   this.game.addMessage('{0} uses {1} to gain {2} gold', this.controller, this, gold);
})
```

I haven't done it because 1. getting promises to work properly in this context will require rewriting most of the pipeline code, 2. it's kinda ugly (though can perhaps make it prettier), and 3. I'd want to do it in a way that supports 'Then' abilities, which also require the pre-then to have succeeded before proceeding.